### PR TITLE
Add pullSecret for quay.io in SA so it's able to get image from priva…

### PR DESCRIPTION
**Pull request by OliviaHY**
### Why:
We will be making quay.io/app sre/gubernator private and use a recently mirrored memcached repo which is private as well. 

### What:
Add pullSecret for quay.io in SA so it's able to get image from private repo.

### CODE REVIEW

File: services/telemeter.libsonnet [hunk 1]
```diff
@@ -130,7 +130,9 @@
         },
       },
     },
-
+    serviceAccount+: {
+      imagePullSecrets+: [{ name: 'quay.io' }],
+    },
```
Comment (onprem) [inline services/telemeter.libsonnet:135]: Looks like this is not working as intended. The telemeter memcached [does not have any service account configured](https://github.com/openshift/telemeter/blob/master/jsonnet/telemeter/server/kubernetes.libsonnet#L197) in it. This is why it is generating an invalid K8s object and the CI is failing.

I think the solution is that we give the Memcached it's own service account, and then we configure pull secrets for it.

Comment (onprem) [reply services/telemeter.libsonnet:135]: I have created a PR for adding a service account for Memcached openshift/telemeter#403. Once it gets merged, we can update the dependency here and regenerate the manifests. CI should pass after this.

Comment (OliviaHY) [reply services/telemeter.libsonnet:135]: That makes sense, thank you for helping @onprem! Is this not an issue with observatorium? Is this [repo](https://github.com/observatorium/observatorium/tree/main/website) where that can be checked?

Comment (OliviaHY) [reply services/telemeter.libsonnet:135]: @onprem I saw the PR for adding the SA is merged, how do we update the dependency here?

**Review by onprem — APPROVED** [review]
This is looking good to go. Thanks @OliviaHY!

**Comment by OliviaHY** [comment]
@periklis I think the changes you quested has been made. Please take another look.

**Review by periklis — APPROVED** [review]
LGTM

**Pull request merged**